src/dvc_data/index.py

```diff
@@ -125,11 +146,24 @@ def iteritems(self, prefix=None, shallow=False):
             self._load(key, entry)
             yield key, entry
 
-    def ls(self, prefix=None):
-        kwargs = {}
-        if prefix:
-            kwargs["prefix"] = prefix
-
+    def info(self, key):
+        try:
+            entry = self[key]
+            return entry.as_dvc_info()
+        except ShortKeyError:
+            return {
+                "type": "directory",
+                "size": 0,
+                "isexec": False,
+                "isdvc": bool(self.longest_prefix(key)),
+                "isout": False,
+                "obj": None,
+                "entry": None,
+            }
+        except KeyError as exc:
+            raise FileNotFoundError from exc
+
+    def ensure_loaded(self, prefix):
```

Comment:
Should it be private?

Reply:
TBH, I only extracted it so pylint wouldn't complain. But it probably should be private since it only makes sense in `.ls()`.